app/components/search_result_component/search_result_component.html.erb

```diff
@@ -0,0 +1,8 @@
+<li>
+  <% case result.class.to_s %>
+  <% when 'User' %>
+    <%= c 'button', label: result.name , styles: [:link], link: user_path(result) %>
+  <% when 'Reply' %>
+    <%= c 'button', label: result.text , styles: [:link], link: user_request_path(result.user, result.request) %>
+  <% end %>
```

Comment:
I think I’d prefer to have a lean template and put that logic into the component class:

```erb
<%= c 'button', label: title , styles: [:link], link: link %>
```

… and then methods in `SearchResult`.